R/db-io.R

```diff
@@ -161,15 +161,15 @@ db_write_table.DBIConnection <- function(con,
                                          values,
                                          temporary = TRUE,
                                          ...) {
-  check_table_ident(table)
+  table <- as_table_ident(table)
   check_character(types, allow_null = TRUE)
   check_named(types)
   check_bool(temporary)
 
   tryCatch(
     dbWriteTable(
       con,
-      name = dbi_quote(table, con),
+      name = table_ident_to_id(table),
```

Comment:
Before we passed a quoted string of class `<SQL>`. It has benefits to instead pass an `<Id>` object (see #1261). As I had to adapt this anyway, I thought I might just change it directly.